### ISSUE TITLE
fatals in docker container when sensors other than coretemp are available

### ISSUE THREAD

**Issue by wez** [automatic-ripping-machine/automatic-ripping-machine#665]
- [x] I have checked the wiki
- [x] I have searched for similar issues

**Describe the bug**

Loading most web pages in the UI fails with an ascii art saying the page does not exist, and the word `coretemp` at the bottom of the screen.

**To Reproduce**

Install docker image, load in browser

### Environment

```
; cat /etc/lsb-release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04.1 LTS"
```

Running `v2_devel` tag from docker from today: https://hub.docker.com/layers/automaticrippingmachine/automatic-ripping-machine/v2_devel/images/sha256-4043ad239254387d1e8cff5c447f6cfa86e22ea7201fdc0e1b770187e92ceeaf?context=explore

### Log file

Omitted; here's the "fix":

The error is caused by this line of code:

```
root@05b8635a53b5:/opt/arm/arm/ui# grep -R coretemp .
./serverutil.py:                self.cpu_temp = temps['coretemp'][0][1]
```

On this system, there is no `coretemp` sensor:

```
; python3
> import psutil
>>> psutil.sensors_temperatures()
{'nvme': [shwtemp(label='Composite', current=52.85, high=84.85, critical=84.85), shwtemp(label='Sensor 1', current=52.85, high=65261.85, critical=65261.85), shwtemp(label='Sensor 2', current=53.85, high=65261.85, critical=65261.85)], 'enp36s0': [shwtemp(label='PHY Temperature', current=78.957, high=None, critical=None), shwtemp(label='MAC Temperature', current=79.086, high=None, critical=None)], 'k10temp': [shwtemp(label='Tctl', current=53.0, high=None, critical=None), shwtemp(label='Tccd1', current=42.25, high=None, critical=None), shwtemp(label='Tccd2', current=36.75, high=None, critical=None)]}
>>>
```

I think the code should be changed to something like:

```python
   self.cpu_temp = 0
   if coretemp := temps.get('coretemp', None):
       self.cpu_temp = coretemp[0][1]
```

**Comment by microtechno9000** [comment]
Hi @wez, thanks for joining the arm community.
Unfortunately at the moment, arm doesn't support Ubuntu 22.04 as there are quite a few breaking changes, one that you just found.

There is some work to go in getting arm supported in 22.04, something that if you can help with you are welcome to fork and start updating the codebase.

However, for the moment if you want to get arm up and running, please use ubuntu 20.04 LTS or one of the other known supported distros.

**Comment by wez** [comment]
FWIW, I don't think this is an ubuntu version issue at all: my hardware doesn't have a `coretemp` sensor. I think that might be an Intel/chipset thing?

Either way, I just edited that file to always return 0 for the temperature to make the docker image work for me.

**Comment by shitwolfymakes** [comment]
@wez You're sure you ran the release in a terminal attached to the docker container, not the host? The `arm-dependencies` image uses a custom 20.04 base image

**Comment by wez** [comment]
I'm running your docker image; ignore the lsb-release bit, that's from the host that is running docker.

**Comment by shitwolfymakes** [comment]
Confirmed the container is 20.04.5 LTS. You must've run that command on your host, in which case the host OS version has no bearing on the container.

@microtechno9000 this'll be for you to fix in the source code 👍

**Comment by microtechno9000** [comment]
@wez I have updated the code, thanks for your code suggestion. I didn't know assignment expression was possible, reading [PEP 572](https://peps.python.org/pep-0572/) was interesting.

Have a look at the commit I made to my fork, if it works for you I will roll into the main ARM repo [#1bf2e1](https://github.com/microtechno9000/automatic-ripping-machine/commit/1bf2e15fce22b97ab267b7bc3edb61c04ea27bbc)

**Comment by wez** [comment]
I patched it in to my docker container and I can confirm that it is working for me, thanks!

**Comment by shitwolfymakes** [comment]
@microtechno9000 Can you open a PR for this that incorporates the fix from my drafted one as well? I can merge that since 1337 is afk and can't review mine atm

**Comment by microtechno9000** [comment]
Closed, issue has been resolved in #671
